### PR TITLE
telemetry: add transaction usage info

### PR DESCRIPTION
**Pull request by youjiali1995**
### What problem does this PR solve?

Problem Summary:

Telemetry reports the usage info of transaction-related features, including async-commit, 1PC and counters of transactions committed with different protocols.

### What is changed and how it works?
What's Changed:

Read values of global sysvars and transaction commit metrics. Based on https://github.com/pingcap/tidb/pull/23454

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

### Release note 

- No release note.

### CODE REVIEW

File: telemetry/data_feature_usage.go [hunk 1]
```diff
@@ -225,20 +227,34 @@ func getTelemetryFeatureUsageInfo(ctx sessionctx.Context) (*featureUsageInfo, er
...
+	if val, err := variable.GetGlobalSystemVar(ctx.GetSessionVars(), variable.TiDBEnableAsyncCommit); err == nil {
+		asyncCommitUsed = val == variable.BoolOn
+	}
+	onePCUsed := false
+	if val, err := variable.GetGlobalSystemVar(ctx.GetSessionVars(), variable.TiDBEnable1PC); err == nil {
+		onePCUsed = val == variable.BoolOn
+	}
+	curr := metrics.GetTxnCommitCounter()
+	diff := curr.Sub(initialTxnCommitCounter)
+	initialTxnCommitCounter = curr
```
Comment (breeswish) [inline telemetry/data_feature_usage.go:258]: This can be called in preview, so that a preview will reset the data.

File: telemetry/data.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ type telemetryData struct {
 	Hardware           []*clusterHardwareItem  `json:"hardware"`
 	Instances          []*clusterInfoItem      `json:"instances"`
 	TelemetryHostExtra *telemetryHostExtraInfo `json:"hostExtra"`
+	FeatureUsage       *featureUsageInfo       `json:"featureUsage"`
```
Comment (youjiali1995) [inline telemetry/data.go:26]: ops, merge conflict.

Comment (lzmhhh123) [inline telemetry/data.go:26]: You can merge FeatureUsageInfo below and it.

**Review by lzmhhh123 — APPROVED** [review]

**Comment by youjiali1995** [comment]
@breeswish PTAL

**Review by breeswish — APPROVED** [review]

**Pull request merged**